lib/openxml/docx/elements/word_processing_shapes_shape_properties.rb

```diff
@@ -0,0 +1,23 @@
+module OpenXml
+  module Docx
+    module Elements
+      class WordProcessingShapesShapeProperties < Container
+        tag :spPr
+        namespace :wps
+
+        attribute :bwMode, expects: :valid_colors
+
+      private
+
+        def valid_colors(value)
+          raise ArgumentError unless ok_values.include? value
+        end
+
+        def ok_values
+          %i(auto black blackGray blackWhite clr gray grayWhite hidden invGray ltGray white)
+        end
```

Comment:
Again, can this work a little more like ActiveRecord validations?

```diff
  module OpenXml
    module Docx
      module Elements
        class WordProcessingShapesShapeProperties < Container
          tag :spPr
          namespace :wps

-         attribute :bwMode, expects: :valid_colors
+         attribute :bwMode, one_of: %i(auto black blackGray blackWhite clr gray grayWhite hidden invGray ltGray white)

-       private
- 
-         def valid_colors(value)
-           raise ArgumentError unless ok_values.include? value
-         end
- 
-         def ok_values
-           %i(auto black blackGray blackWhite clr gray grayWhite hidden invGray ltGray white)
-         end
- 
        end
      end
    end
  end
```

That'd work if we just have `attribute` (the class method) accept options like `:one_of` or `:matches` and use them in the `define_method`.